src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -2735,7 +2772,28 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
                             x.base.base.loc, false);
         loop_end = ASRUtils::EXPR(tmp);
         ASR::do_loop_head_t head;
-        head.m_v = target;
+
+        if(is_explicit_iterator_required) {
+            body.reserve(al, x.n_body + 1);
+            // add an assignment instruction to body to assign value of loop_src_var at an index to the loop_target_var
+            auto explicit_iter_var = ASR::make_Var_t(al, x.base.base.loc, current_scope->get_symbol("__explicit_iterator"));
+            auto index_plus_one = ASR::make_IntegerBinOp_t(al, x.base.base.loc, ASRUtils::EXPR(explicit_iter_var),
+                ASR::binopType::Add, constant_one, a_type, nullptr);
```

Comment:
I am needing to add `one` to the current `index`, that is, to add `one` to the current value of `__explicit_iterator`. Here (lines `2780-2781`), I am using `BinOp` to add the needed `one`.  Without adding one, it seems while indexing the `__explicit_iterator` fells short by `one` unit.

**doubt:** Please, could someone possibly share what might be the issue for `__explicit_iterator` felling short/behind by `one` unit?

Reply:
Indexing string require addition of one (because of legacy reasons I believe, we need to fix this but we can do it later). For now its fine. So, I'd say don't worry about adding 1.

Reply:
Got it. Thank you.